code/modules/clothing/head/hardhat.dm

```diff
@@ -11,43 +11,61 @@
 	siemens_coefficient = 0.9
 	flags_inventory = BLOCKSHARPOBJ
 
-	attack_self(mob/user)
-		if(!isturf(user.loc))
-			to_chat(user, "You cannot turn the light on while in [user.loc]")
-			return
-		on = !on
-		icon_state = "hardhat[on]_[hardhat_color]"
-		item_state = "hardhat[on]_[hardhat_color]"
+/obj/item/clothing/head/hardhat/attack_self(mob/user)
+	if(!isturf(user.loc))
+		to_chat(user, "You cannot turn the light on while in [user.loc]")
+		return
+	on = !on
+	icon_state = "hardhat[on]_[hardhat_color]"
+	item_state = "hardhat[on]_[hardhat_color]"
 
-		if(on)	user.SetLuminosity(brightness_on)
-		else	user.SetLuminosity(-brightness_on)
+	if(user == loc)
+		var/mob/M = loc
+		M.update_inv_head()
 
-		if(ismob(loc))
-			var/mob/M = loc
-			M.update_inv_head()
+	update_brightness()
+	update_action_button_icons()
 
+/obj/item/clothing/head/hardhat/proc/turn_off_light(mob/bearer)
+	if(on)
+		on = FALSE
+		update_brightness(bearer)
 		update_action_button_icons()
+		return TRUE
+	return FALSE
 
-	pickup(mob/user)
-		if(on)
+/obj/item/clothing/head/hardhat/proc/update_brightness(var/mob/user = null)
+	if(!user && ismob(loc))
```

Comment:
To be fair this shouldn't be needed. We should look instead when `update_brightness` is being called without a user and fix that instead.